### DISCUSSION TITLE
Adding attachments

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#6222] [Help]
I am looking to upload a invoices/recepits for each customer.

I have seen that there is an attachment widget to upload files and watched the video [Here](https://docs.budibase.com/docs/attachments)

The issue I am having is... I want to upload invoices/recepits for each customer and link it to their account so that they can be retreived when they log in.

I have created a new table via the BudiBase back end and added the relevant columns... set up the relationship between this table and my customer accounts table but then realised that there was no `attachment` option available as a column type.

I then had a play about and seen that this option is only available via the Budibase DB. I then set up the same table here but am not able to use the relationship function to link the invoices/recepits to customers.

Is there a way around this?

**Comment by melohagan** [comment]
Hi @PaulMcF1987 

There is a feature request open for supporting Attachment types in external datasources: https://github.com/Budibase/budibase/issues/5486

Would you be able to clarify this please?
> I then set up the same table here but am not able to use the relationship function to link the invoices/recepits to customers.

**Reply by PaulMcF1987** [reply]
@melohagan 
Rather than trying to set up the table under my own MYSQL DB I tried setting it up via `Budibase DB` however I wasnt able to create relationships between tables in Budibase DB and MYSQL.

Is there a way That I could use this, setting up my own MSQL table? If I se HeidiSQL and change the Datatype to Blob, for example?

**Reply by melohagan** [reply]
You cannot have direct relationships between different datasources. 

What you could try is to have a "soft relationship" between the Budibase DB attachment table and your MySQL customer table.

In this case, you would not use the standard relationship functionality, however you could have a screen where you pass the id in the URL and then filter on that id.
Guide here: https://docs.budibase.com/docs/url-parameters
